pkg/policycache/cache.go

```diff
@@ -62,31 +63,56 @@ func (c *cache) GetPolicies(pkey PolicyType, gvr schema.GroupVersionResource, su
 func filterPolicies(pkey PolicyType, result []kyvernov1.PolicyInterface, nspace string) []kyvernov1.PolicyInterface {
 	var policies []kyvernov1.PolicyInterface
 	for _, policy := range result {
+		var filteredPolicy kyvernov1.PolicyInterface
 		keepPolicy := true
 		switch pkey {
 		case ValidateAudit:
-			keepPolicy = checkValidationFailureActionOverrides(false, nspace, policy)
+			keepPolicy, filteredPolicy = checkValidationFailureActionOverrides(false, nspace, policy)
 		case ValidateEnforce:
-			keepPolicy = checkValidationFailureActionOverrides(true, nspace, policy)
+			keepPolicy, filteredPolicy = checkValidationFailureActionOverrides(true, nspace, policy)
 		}
 		// add policy to result
 		if keepPolicy {
-			policies = append(policies, policy)
+			policies = append(policies, filteredPolicy)
 		}
 	}
 	return policies
 }
 
-func checkValidationFailureActionOverrides(enforce bool, ns string, policy kyvernov1.PolicyInterface) bool {
-	validationFailureAction := policy.GetSpec().GetValidationFailureAction()
-	validationFailureActionOverrides := policy.GetSpec().GetValidationFailureActionOverrides()
-	if validationFailureAction.Enforce() != enforce && (ns == "" || len(validationFailureActionOverrides) == 0) {
-		return false
-	}
-	for _, action := range validationFailureActionOverrides {
-		if action.Action.Enforce() != enforce && wildcard.CheckPatterns(action.Namespaces, ns) {
-			return false
+func checkValidationFailureActionOverrides(enforce bool, ns string, policy kyvernov1.PolicyInterface) (bool, kyvernov1.PolicyInterface) {
+	var filteredRules []kyvernov1.Rule
+	for _, rule := range autogen.ComputeRules(policy, "") {
+		if !rule.HasValidate() {
```

Comment:
What about `imageVerify` rule?

Reply:
`imageVerify` rules don't have the validationFailureActionOverrides.

This function is called to filter the `validate` policies only based on its action and the overrides.

https://github.com/kyverno/kyverno/blob/cf1ac712921f39f1500b4b4dc8191a0c164cb0dc/pkg/policycache/cache.go#L69-L72

That's because we handle audit and enforce validations separately but for verifyImage rules, we only check the `ValidationFailureAction`.